rule-templates/ruby/base-rules.mdc

```diff
@@ -1,5 +1,256 @@
 ---
-description:
-globs:
-alwaysApply: false
+description: 
+globs: 
+alwaysApply: true
 ---
+## Project Structure Standards
+
+### Directory Organization
+Organize code with the following structure:
+- `app/controllers/api/v1` for versioned API controllers
+- `app/models` for data models
+- `app/services` for business logic
+- `app/policies` for authorization rules
+- `app/serializers` for JSON response formatting
+- `app/jobs` for background processing with GoodJob
+- `app/forms` for form objects using YAAF
+- `app/decorators` for view-related logic using Draper
+
+### Code Quality Anti-patterns to Avoid
+- **Fat Models:** Models with too much business logic - move to service objects or POROs
+- **God Classes:** Classes doing too much - break into smaller, focused classes
+- **Duplicated Code:** Repeated code - follow DRY principle and extract to shared methods
+- **Long Methods:** Complex methods - break into smaller, focused methods
+- **Magic Numbers:** Hardcoded values - use constants or enums instead
+
+## API Conventions
+- Use versioning in URLs: `/api/v1/resource`
+- Maintain backward compatibility between versions
+- Use appropriate HTTP status codes
+- Keep consistent response format
+- Include relevant metadata in responses
+- Write tests for all API endpoints
+- Maintain high code coverage
+- Avoid using callbacks in models
+- Use namespaces for business areas (e.g., "payments")
```

Comment:
I feel like some of these rules should be in a more generic section rather than "API Conventions"